tests/modules/configs/test_config.py

```diff
@@ -0,0 +1,91 @@
+import os
+import pytest
+import sys
+
+from jsonschema import ValidationError
+
+sys.path.append(os.getcwd())
+
+from ingestation.modules.configs.config import validate_json
+from ingestation.modules.configs.schemas import application, project, operator
+from tests.data.configs import validate_json_passed, validate_json_failed, err_msg
+
+
```

Comment:
Let's aggregate all those functions under a class e.g. `TestAllConfTypes`. Then, you can move parametrization to class-level e.g.
```
@pytest.mark.parametrize("conf_type,schema", [
    ["application", application],
    ["project", project],
    ["operator", operator]
])
class TestAllConfigTypes:

    def test_validate_json_unrecognized_key(self, conf_type: str, schema):
        full_conf = getattr(validate_json_failed, conf_type)
        conf = full_conf.get("unrecognized_key")

        with pytest.raises(ValidationError) as ex:
            validate_json(conf, schema)
        assert err_msg.validate_json.get("unrecognized_key") in str(ex.value)
```

Reply:
Besides `tests.modules.configs.test_config.TestConf` should come first in this commit. `TestAllConfTypes` should be committed only after each of their code is implemented.

Reply:
This work too:
```
@pytest.mark.parametrize("conf_type,schema", [
    ["application", application],
    ["project", project],
    ["operator", operator]
])
class TestAllConfigTypes:

    @pytest.mark.parametrize("validation_error", ["invalid_data_type", "mandatory_key",
        "too_long_value", "unrecognized_key"])
    def test_validate_json_failed(self, conf_type: str, schema, validation_error: str):
        full_conf = getattr(validate_json_failed, conf_type)
        conf = full_conf.get(validation_error)

        with pytest.raises(ValidationError) as ex:
            validate_json(conf, schema)
        assert err_msg.validate_json.get(validation_error) in str(ex.value)

    def test_validate_json_non_mandatory_key(self, conf_type: str, schema):
        full_conf = getattr(validate_json_passed, conf_type)
        conf = full_conf.get("non_mandatory_key")

        assert validate_json(conf, schema) is None

    def test_validate_json_passed(self, conf_type: str, schema):
        full_conf = getattr(validate_json_passed, conf_type)
        conf = full_conf.get("validation_passed")

        assert validate_json(conf, schema) is None
```